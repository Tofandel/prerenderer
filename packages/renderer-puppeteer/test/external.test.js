const path = require('path')
const Prerenderer = require('@prerenderer/prerenderer')
const Renderer = require('@prerenderer/renderer-puppeteer')

const testDir = path.resolve(__dirname, '../../../tests/external')

test('renders routes with externally loaded script and encoding', async () => {
  const prerenderer = new Prerenderer({
    staticDir: testDir,
    renderer: new Renderer(),
  })

  await prerenderer.initialize()
  const renderedRoutes = await prerenderer.renderRoutes(['/external_script.html', '/external_scriptÃ©.html'])
  await prerenderer.destroy()

  expect(renderedRoutes).toMatchSnapshot()
})
