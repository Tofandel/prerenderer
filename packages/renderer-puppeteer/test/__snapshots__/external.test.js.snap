// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders routes with externally loaded script and encoding 1`] = `
[
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: External script</title>
  <script src="assets/script.js"></script>
</head>
<body>


<p>Render Output</p></body></html>",
    "originalRoute": "/external_script.html",
    "route": "/external_script.html",
  },
  {
    "html": "<!DOCTYPE html><html><head>
  <title>Prerenderer Test: External script with unicode</title>
  <script src="assets/scripté.js"></script>
</head>
<body>


<p>Renderé the Outputé</p></body></html>",
    "originalRoute": "/external_scripté.html",
    "route": "/external_scripté.html",
  },
]
`;
